src/Microsoft.Dnx.Tooling/List/MismatchDependencyRenderer.cs

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.Versioning;
+using System.Text;
+using Microsoft.Dnx.Runtime;
+using Microsoft.Dnx.Tooling.Algorithms;
+using NuGet;
+
+namespace Microsoft.Dnx.Tooling.List
+{
+    internal class MismatchDependencyRenderer
+    {
+        private readonly DependencyListOptions _options;
+        private readonly FrameworkName _framework;
+
+        public MismatchDependencyRenderer(DependencyListOptions options, FrameworkName framework)
+        {
+            _options = options;
+            _framework = framework;
+        }
+
+        public void Render(IGraphNode<LibraryDependency> root)
+        {
+            // tuples of <Library Name, Requested Version, Actual Version>
+            var results = new HashSet<Tuple<string, string, string>>();
+
+            root.DepthFirstPreOrderWalk(
+                (node, ancestors) =>
+                {
+                    var dependency = node.Item;
+                    if (IsLibraryMismatch(dependency))
+                    {
+                        results.Add(Tuple.Create(
+                            dependency.Library.Identity.Name,
+                            dependency.LibraryRange.VersionRange?.MinVersion.ToString(),
+                            dependency.Library.Identity.Version?.ToString()));
+                    }
+
+                    return true;
+                });
+
+            if (results.Any())
+            {
+                var format = GetFormat(results, padding: 2);
+
+                RenderTitle(format);
+                RenderMismatches(format, results);
+            }
+        }
+
+        private void RenderTitle(string format)
+        {
+            _options.Reports.WriteInformation($"\n[Target framework {_framework} ({VersionUtility.GetShortFrameworkName(_framework)})]\n");
+            _options.Reports.WriteInformation(string.Format(format, "Dependency", "Requested", "Resolved"));
+            _options.Reports.WriteInformation(string.Format(format, "----------", "---------", "--------"));
+        }
+
+        private void RenderMismatches(string format, HashSet<Tuple<string, string, string>> results)
+        {
+            foreach (var row in results.OrderBy(r => r.Item1))
+            {
+                _options.Reports.WriteInformation(string.Format(format, row.Item1, row.Item2, row.Item3));
+            }
+        }
+
+        private static string GetFormat(IEnumerable<Tuple<string, string, string>> data, int padding)
```

Comment:
Good point. But given the incoming refactory, I'd like to hold off for a bit longer.